ui/pages/Project/components/RnaSeqResultPage.jsx

```diff
@@ -0,0 +1,146 @@
+    this.setState({ tissueType: data.value })
+  }
+
+  render() {
+    const { familyGuid, rnaSeqData, significantJunctionOutliers, genesById, tissueOptions } = this.props
+    const { tissueType } = this.state
+    const showTissueType = tissueType || (tissueOptions?.length > 0 ? tissueOptions[0].value : null)
+    const tissueDisplay = TISSUE_DISPLAY[showTissueType]
+
+    const outlierPlotConfigs = OUTLIER_VOLCANO_PLOT_CONFIGS.map(({ formatData, ...config }) => {
+      const data = rnaSeqData[config.key] || []
+      return ({ data: formatData ? formatData(data, showTissueType) : data, ...config })
+    }).filter(({ data }) => data.length)
+
+    const tableData = significantJunctionOutliers.reduce(
+      (acc, outlier) => (outlier.tissueType === showTissueType ? [...acc, outlier] : acc), [],
+    )
+
+    return (
+      <div>
+        {showTissueType && (
+          <TissueContainer>
+            Tissue type: &nbsp;
+            {tissueOptions.length > 1 ? (
+              <Dropdown
+                text={tissueDisplay || 'Unknown Tissue'}
+                value={showTissueType}
+                options={tissueOptions}
+                onChange={this.onTissueChange}
+              />
+            ) : tissueDisplay}
+          </TissueContainer>
+        )}
+        { outlierPlotConfigs.length > 0 && (
+          <React.Suspense fallback={<Loader />}>
+            <Grid>
+              <Grid.Row divided columns={outlierPlotConfigs.length}>
+                {outlierPlotConfigs.map(({ key, data, ...config }) => (
+                  <Grid.Column key={key} width={8}>
+                    <RnaSeqOutliers
+                      familyGuid={familyGuid}
+                      rnaSeqData={data}
+                      genesById={genesById}
+                      {...config}
+                    />
+                  </Grid.Column>
+                ))}
+              </Grid.Row>
+            </Grid>
+          </React.Suspense>
+        )}
+        {(tableData.length > 0) && (
+          <FamilyReads
+            layout={RnaSeqJunctionOutliersTable}
+            noTriggerButton
+            data={tableData}
+            defaultSortColumn="pValue"
+            maxHeight="600px"
+          />
+        )}
+      </div>
+    )
+  }
+
+}
+
+const RnaSeqResultPage = React.memo(({ individual, rnaSeqData, load, loading, ...props }) => (
+  <DataLoader content={rnaSeqData} contentId={individual.individualGuid} load={load} loading={loading}>
+    <BaseRnaSeqResultPage familyGuid={individual.familyGuid} rnaSeqData={rnaSeqData} {...props} />
+  </DataLoader>
+))
+
+RnaSeqResultPage.propTypes = {
+  individual: PropTypes.object,
+  rnaSeqData: PropTypes.object,
+  load: PropTypes.func,
+  loading: PropTypes.bool,
+}
+
+const mapStateToProps = (state, ownProps) => ({
+  individual: getIndividualsByGuid(state)[ownProps.match.params.individualGuid],
+  rnaSeqData: Object.entries(getRnaSeqDataByIndividual(state)[ownProps.match.params.individualGuid] || {}).reduce(
```

Comment:
it is important to use selectors when computing Arrays/ Objects as props, as this prevents unneccessary re-renders. Because you are computing this here, every time this function is called the prop will change and the page will render even if no actual data changes. I don't think this formatting is neccessary here at all, I think its fine to use the existing selector for the prop `getRnaSeqDataByIndividual(state)[ownProps.match.params.individualGuid]` and then in the render method do the flattening

Reply:
The problem is for the expression outliers. When the users change the tissue type, the expression outlier data shouldn't change. But if we use `Object.entries()` to get the data in the render method, the data will be new each time rerendering which causes the volcano plot to be re-plotted. It is better for the splice junction outliers because it always requires re-plotting when tissue type changes.

Reply:
okay, but having the computation here is also leading to unnecessary re-renders. The solution would be that you create a new selector to use here instead

Reply:
this blocks approval

Reply:
Oh, I changed it in the other branch.